File: OPENAPI-REST-API/openapi-client/cpp-tiny/lib/Models/DomainServiceInfo.cpp


#include "DomainServiceInfo.h"

using namespace Tiny;

DomainServiceInfo::DomainServiceInfo()
{
	domain_id = std::string();
	domain_hostname = std::string();
	domain_username = std::string();
	domain_password = std::string();
	domain_type = std::string();
	domain_currency = std::string();
	domain_expire_date = std::string();
	domain_order_date = std::string();
	domain_custid = std::string();
	domain_status = std::string();
	domain_invoice = std::string();
	domain_coupon = std::string();
}

DomainServiceInfo::DomainServiceInfo(std::string jsonString)
{
	this->fromJson(jsonString);
}

DomainServiceInfo::~DomainServiceInfo()
{

}

void
DomainServiceInfo::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *domain_idKey = "domain_id";

    if(object.has_key(domain_idKey))
    {
        bourne::json value = object[domain_idKey];



        jsonToValue(&domain_id, value, "std::string");


    }

    const char *domain_hostnameKey = "domain_hostname";

    if(object.has_key(domain_hostnameKey))
    {
        bourne::json value = object[domain_hostnameKey];



        jsonToValue(&domain_hostname, value, "std::string");


    }

    const char *domain_usernameKey = "domain_username";

    if(object.has_key(domain_usernameKey))
    {
        bourne::json value = object[domain_usernameKey];



        jsonToValue(&domain_username, value, "std::string");


    }

    const char *domain_passwordKey = "domain_password";

    if(object.has_key(domain_passwordKey))
    {
        bourne::json value = object[domain_passwordKey];



        jsonToValue(&domain_password, value, "std::string");


    }

    const char *domain_typeKey = "domain_type";

    if(object.has_key(domain_typeKey))
    {
        bourne::json value = object[domain_typeKey];



        jsonToValue(&domain_type, value, "std::string");


    }

    const char *domain_currencyKey = "domain_currency";

    if(object.has_key(domain_currencyKey))
    {
        bourne::json value = object[domain_currencyKey];



        jsonToValue(&domain_currency, value, "std::string");


    }

    const char *domain_expire_dateKey = "domain_expire_date";

    if(object.has_key(domain_expire_dateKey))
    {
        bourne::json value = object[domain_expire_dateKey];



        jsonToValue(&domain_expire_date, value, "std::string");


    }

    const char *domain_order_dateKey = "domain_order_date";

    if(object.has_key(domain_order_dateKey))
    {
        bourne::json value = object[domain_order_dateKey];



        jsonToValue(&domain_order_date, value, "std::string");


    }

    const char *domain_custidKey = "domain_custid";

    if(object.has_key(domain_custidKey))
    {
        bourne::json value = object[domain_custidKey];



        jsonToValue(&domain_custid, value, "std::string");


    }

    const char *domain_statusKey = "domain_status";

    if(object.has_key(domain_statusKey))
    {
        bourne::json value = object[domain_statusKey];



        jsonToValue(&domain_status, value, "std::string");


    }

    const char *domain_invoiceKey = "domain_invoice";

    if(object.has_key(domain_invoiceKey))
    {
        bourne::json value = object[domain_invoiceKey];



        jsonToValue(&domain_invoice, value, "std::string");


    }

    const char *domain_couponKey = "domain_coupon";

    if(object.has_key(domain_couponKey))
    {
        bourne::json value = object[domain_couponKey];



        jsonToValue(&domain_coupon, value, "std::string");


    }


}

bourne::json
DomainServiceInfo::toJson()
{
    bourne::json object = bourne::json::object();





    object["domain_id"] = getDomainId();






    object["domain_hostname"] = getDomainHostname();






    object["domain_username"] = getDomainUsername();






    object["domain_password"] = getDomainPassword();






    object["domain_type"] = getDomainType();






    object["domain_currency"] = getDomainCurrency();






    object["domain_expire_date"] = getDomainExpireDate();






    object["domain_order_date"] = getDomainOrderDate();






    object["domain_custid"] = getDomainCustid();






    object["domain_status"] = getDomainStatus();






    object["domain_invoice"] = getDomainInvoice();






    object["domain_coupon"] = getDomainCoupon();



    return object;

}

std::string
DomainServiceInfo::getDomainId()
{
	return domain_id;
}

void
DomainServiceInfo::setDomainId(std::string domain_id)
{
	this->domain_id = domain_id;
}

std::string
DomainServiceInfo::getDomainHostname()
{
	return domain_hostname;
}

void
DomainServiceInfo::setDomainHostname(std::string domain_hostname)
{
	this->domain_hostname = domain_hostname;
}

std::string
DomainServiceInfo::getDomainUsername()
{
	return domain_username;
}

void
DomainServiceInfo::setDomainUsername(std::string domain_username)
{
	this->domain_username = domain_username;
}

std::string
DomainServiceInfo::getDomainPassword()
{
	return domain_password;
}

void
DomainServiceInfo::setDomainPassword(std::string domain_password)
{
	this->domain_password = domain_password;
}

std::string
DomainServiceInfo::getDomainType()
{
	return domain_type;
}

void
DomainServiceInfo::setDomainType(std::string domain_type)
{
	this->domain_type = domain_type;
}

std::string
DomainServiceInfo::getDomainCurrency()
{
	return domain_currency;
}

void
DomainServiceInfo::setDomainCurrency(std::string domain_currency)
{
	this->domain_currency = domain_currency;
}

std::string
DomainServiceInfo::getDomainExpireDate()
{
	return domain_expire_date;
}

void
DomainServiceInfo::setDomainExpireDate(std::string domain_expire_date)
{
	this->domain_expire_date = domain_expire_date;
}

std::string
DomainServiceInfo::getDomainOrderDate()
{
	return domain_order_date;
}

void
DomainServiceInfo::setDomainOrderDate(std::string domain_order_date)
{
	this->domain_order_date = domain_order_date;
}

std::string
DomainServiceInfo::getDomainCustid()
{
	return domain_custid;
}

void
DomainServiceInfo::setDomainCustid(std::string domain_custid)
{
	this->domain_custid = domain_custid;
}

std::string
DomainServiceInfo::getDomainStatus()
{
	return domain_status;
}

void
DomainServiceInfo::setDomainStatus(std::string domain_status)
{
	this->domain_status = domain_status;
}

std::string
DomainServiceInfo::getDomainInvoice()
{
	return domain_invoice;
}

void
DomainServiceInfo::setDomainInvoice(std::string domain_invoice)
{
	this->domain_invoice = domain_invoice;
}

std::string
DomainServiceInfo::getDomainCoupon()
{
	return domain_coupon;
}

void
DomainServiceInfo::setDomainCoupon(std::string domain_coupon)
{
	this->domain_coupon = domain_coupon;
}

File: OPENAPI-REST-API/swagger-client/spring/src/main/java/io/swagger/model/DomainServiceInfo.java
package io.swagger.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.v3.oas.annotations.media.Schema;
import org.springframework.validation.annotation.Validated;
import org.openapitools.jackson.nullable.JsonNullable;
import io.swagger.configuration.NotUndefined;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonSetter;
import com.fasterxml.jackson.annotation.Nulls;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * Detailed domain service record for a domain order.
 */
@Schema(description = "Detailed domain service record for a domain order.")
@Validated
@NotUndefined



public class DomainServiceInfo   {
  @JsonProperty("domain_id")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainId = null;

  @JsonProperty("domain_hostname")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainHostname = null;

  @JsonProperty("domain_username")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainUsername = null;

  @JsonProperty("domain_password")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainPassword = null;

  @JsonProperty("domain_type")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainType = null;

  @JsonProperty("domain_currency")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainCurrency = null;

  @JsonProperty("domain_expire_date")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainExpireDate = null;

  @JsonProperty("domain_order_date")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainOrderDate = null;

  @JsonProperty("domain_custid")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainCustid = null;

  @JsonProperty("domain_status")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainStatus = null;

  @JsonProperty("domain_invoice")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainInvoice = null;

  @JsonProperty("domain_coupon")

  @JsonInclude(JsonInclude.Include.NON_ABSENT)  // Exclude from JSON if absent
  @JsonSetter(nulls = Nulls.FAIL)    // FAIL setting if the value is null
  private String domainCoupon = null;


  public DomainServiceInfo domainId(String domainId) { 

    this.domainId = domainId;
    return this;
  }

  /**
   * Get domainId
   * @return domainId
   **/
  
  @Schema(description = "")
  
  public String getDomainId() {  
    return domainId;
  }



  public void setDomainId(String domainId) { 
    this.domainId = domainId;
  }

  public DomainServiceInfo domainHostname(String domainHostname) { 

    this.domainHostname = domainHostname;
    return this;
  }

  /**
   * Get domainHostname
   * @return domainHostname
   **/
  
  @Schema(description = "")
  
  public String getDomainHostname() {  
    return domainHostname;
  }



  public void setDomainHostname(String domainHostname) { 
    this.domainHostname = domainHostname;
  }

  public DomainServiceInfo domainUsername(String domainUsername) { 

    this.domainUsername = domainUsername;
    return this;
  }

  /**
   * Get domainUsername
   * @return domainUsername
   **/
  
  @Schema(description = "")
  
  public String getDomainUsername() {  
    return domainUsername;
  }



  public void setDomainUsername(String domainUsername) { 
    this.domainUsername = domainUsername;
  }

  public DomainServiceInfo domainPassword(String domainPassword) { 

    this.domainPassword = domainPassword;
    return this;
  }

  /**
   * Get domainPassword
   * @return domainPassword
   **/
  
  @Schema(description = "")
  
  public String getDomainPassword() {  
    return domainPassword;
  }



  public void setDomainPassword(String domainPassword) { 
    this.domainPassword = domainPassword;
  }

  public DomainServiceInfo domainType(String domainType) { 

    this.domainType = domainType;
    return this;
  }

  /**
   * Get domainType
   * @return domainType
   **/
  
  @Schema(description = "")
  
  public String getDomainType() {  
    return domainType;
  }



  public void setDomainType(String domainType) { 
    this.domainType = domainType;
  }

  public DomainServiceInfo domainCurrency(String domainCurrency) { 

    this.domainCurrency = domainCurrency;
    return this;
  }

  /**
   * Get domainCurrency
   * @return domainCurrency
   **/
  
  @Schema(description = "")
  
  public String getDomainCurrency() {  
    return domainCurrency;
  }



  public void setDomainCurrency(String domainCurrency) { 
    this.domainCurrency = domainCurrency;
  }

  public DomainServiceInfo domainExpireDate(String domainExpireDate) { 

    this.domainExpireDate = domainExpireDate;
    return this;
  }

  /**
   * Get domainExpireDate
   * @return domainExpireDate
   **/
  
  @Schema(description = "")
  
  public String getDomainExpireDate() {  
    return domainExpireDate;
  }



  public void setDomainExpireDate(String domainExpireDate) { 
    this.domainExpireDate = domainExpireDate;
  }

  public DomainServiceInfo domainOrderDate(String domainOrderDate) { 

    this.domainOrderDate = domainOrderDate;
    return this;
  }

  /**
   * Get domainOrderDate
   * @return domainOrderDate
   **/
  
  @Schema(description = "")
  
  public String getDomainOrderDate() {  
    return domainOrderDate;
  }



  public void setDomainOrderDate(String domainOrderDate) { 
    this.domainOrderDate = domainOrderDate;
  }

  public DomainServiceInfo domainCustid(String domainCustid) { 

    this.domainCustid = domainCustid;
    return this;
  }

  /**
   * Get domainCustid
   * @return domainCustid
   **/
  
  @Schema(description = "")
  
  public String getDomainCustid() {  
    return domainCustid;
  }



  public void setDomainCustid(String domainCustid) { 
    this.domainCustid = domainCustid;
  }

  public DomainServiceInfo domainStatus(String domainStatus) { 

    this.domainStatus = domainStatus;
    return this;
  }

  /**
   * Get domainStatus
   * @return domainStatus
   **/
  
  @Schema(description = "")
  
  public String getDomainStatus() {  
    return domainStatus;
  }



  public void setDomainStatus(String domainStatus) { 
    this.domainStatus = domainStatus;
  }

  public DomainServiceInfo domainInvoice(String domainInvoice) { 

    this.domainInvoice = domainInvoice;
    return this;
  }

  /**
   * Get domainInvoice
   * @return domainInvoice
   **/
  
  @Schema(description = "")
  
  public String getDomainInvoice() {  
    return domainInvoice;
  }



  public void setDomainInvoice(String domainInvoice) { 
    this.domainInvoice = domainInvoice;
  }

  public DomainServiceInfo domainCoupon(String domainCoupon) { 

    this.domainCoupon = domainCoupon;
    return this;
  }

  /**
   * Get domainCoupon
   * @return domainCoupon
   **/
  
  @Schema(description = "")
  
  public String getDomainCoupon() {  
    return domainCoupon;
  }



  public void setDomainCoupon(String domainCoupon) { 
    this.domainCoupon = domainCoupon;
  }

  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    DomainServiceInfo domainServiceInfo = (DomainServiceInfo) o;
    return Objects.equals(this.domainId, domainServiceInfo.domainId) &&
        Objects.equals(this.domainHostname, domainServiceInfo.domainHostname) &&
        Objects.equals(this.domainUsername, domainServiceInfo.domainUsername) &&
        Objects.equals(this.domainPassword, domainServiceInfo.domainPassword) &&
        Objects.equals(this.domainType, domainServiceInfo.domainType) &&
        Objects.equals(this.domainCurrency, domainServiceInfo.domainCurrency) &&
        Objects.equals(this.domainExpireDate, domainServiceInfo.domainExpireDate) &&
        Objects.equals(this.domainOrderDate, domainServiceInfo.domainOrderDate) &&
        Objects.equals(this.domainCustid, domainServiceInfo.domainCustid) &&
        Objects.equals(this.domainStatus, domainServiceInfo.domainStatus) &&
        Objects.equals(this.domainInvoice, domainServiceInfo.domainInvoice) &&
        Objects.equals(this.domainCoupon, domainServiceInfo.domainCoupon);
  }

  @Override
  public int hashCode() {
    return Objects.hash(domainId, domainHostname, domainUsername, domainPassword, domainType, domainCurrency, domainExpireDate, domainOrderDate, domainCustid, domainStatus, domainInvoice, domainCoupon);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class DomainServiceInfo {\n");
    
    sb.append("    domainId: ").append(toIndentedString(domainId)).append("\n");
    sb.append("    domainHostname: ").append(toIndentedString(domainHostname)).append("\n");
    sb.append("    domainUsername: ").append(toIndentedString(domainUsername)).append("\n");
    sb.append("    domainPassword: ").append(toIndentedString(domainPassword)).append("\n");
    sb.append("    domainType: ").append(toIndentedString(domainType)).append("\n");
    sb.append("    domainCurrency: ").append(toIndentedString(domainCurrency)).append("\n");
    sb.append("    domainExpireDate: ").append(toIndentedString(domainExpireDate)).append("\n");
    sb.append("    domainOrderDate: ").append(toIndentedString(domainOrderDate)).append("\n");
    sb.append("    domainCustid: ").append(toIndentedString(domainCustid)).append("\n");
    sb.append("    domainStatus: ").append(toIndentedString(domainStatus)).append("\n");
    sb.append("    domainInvoice: ").append(toIndentedString(domainInvoice)).append("\n");
    sb.append("    domainCoupon: ").append(toIndentedString(domainCoupon)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
